core/src/main/kotlin/evaluation/service/EvService.kt

```diff
@@ -0,0 +1,128 @@
+import java.net.URLEncoder
+
+@Service
+class EvService(
+    private val snuttEvWebClient: SnuttEvWebClient,
+    private val timetableService: TimetableService,
+    private val coursebookService: CoursebookService,
+    private val userService: UserService,
+) {
+    suspend fun handleRouting(
+        userId: String,
+        requestPath: String,
+        requestQueryParams: MultiValueMap<String, String> = buildMultiValueMap(mapOf()),
+        originalBody: String,
+        method: HttpMethod,
+    ): Map<String, Any?> {
+        val result: MutableMap<String, Any?> =
+            snuttEvWebClient.method(method)
+                .uri { builder -> builder.path(requestPath).queryParams(requestQueryParams).build() }
+                .header("Snutt-User-Id", userId)
+                .header(HttpHeaders.CONTENT_ENCODING, "UTF-8")
+                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                .body(BodyInserters.fromValue(originalBody))
+                .retrieve()
+                .onStatus(HttpStatusCode::isError) { response ->
+                    response.bodyToMono<Map<String, Any?>>()
+                        .flatMap { errorBody ->
+                            Mono.error(ProxyException(response.statusCode(), errorBody))
+                        }
+                }
+                .bodyToMono<MutableMap<String, Any?>>()
+                .awaitSingle()
+        return updateUserInfo(result)
+    }
+
+    suspend fun getMyLatestLectures(
+        userId: String,
+        requestQueryParams: MultiValueMap<String, String>? = null,
+    ): Map<String, Any?> {
+        val recentLectures: List<EvLectureInfoDto> =
+            coursebookService.getLastTwoCourseBooksBeforeCurrent().flatMap { coursebook ->
+                timetableService.getTimetablesBySemester(userId, coursebook.year, coursebook.semester)
+                    .toList()
+                    .flatMap { timetable ->
+                        timetable.lectures.map { lecture ->
+                            EvLectureInfoDto(
+                                lecture,
+                                coursebook.year,
+                                coursebook.semester,
+                            )
+                        }
+                    }
+            }
+
+        val encodedJson =
+            withContext(Dispatchers.IO) {
+                URLEncoder.encode(
+                    ObjectMapper().setPropertyNamingStrategy(PropertyNamingStrategies.SNAKE_CASE).writeValueAsString(recentLectures),
+                    "UTF-8",
+                )
+            }
+
+        return snuttEvWebClient.get()
+            .uri { builder ->
+                UriComponentsBuilder.fromUri(builder.build())
+                    .path("/v1/users/me/lectures/latest")
+                    .queryParam("snutt_lecture_info", encodedJson)
+                    .queryParams(requestQueryParams)
+                    .build(true).toUri()
+            }
```

Comment:
webclient 빌더 문법 저도 싫어하는데
Encoder 가 이상하게 쌓여있는게 좀 더 싫어서 아래 코드가 나아보입니다.

```suggestion
        val lectureInfoParam =  ObjectMapper().setPropertyNamingStrategy(PropertyNamingStrategies.SNAKE_CASE).writeValueAsString(recentLectures)

        return snuttEvWebClient.get()
            .uri { builder ->
                builder
                    .path("/v1/users/me/lectures/latest")
                    .queryParam("snutt_lecture_info", "{lectureInfoParam}")
                    .queryParams(requestQueryParams)
                    .build(lectureInfoParam)
            }
```